File: DSA/JavaScript/BinaryTree/BinaryTree.js
class Node {
	constructor(value) {
		this.value = value;
		this.left = null;
		this.right = null;
	}
}

function createBT() {
	// Tree Created
	//       1
	//      / \
	//     2   3
	//    / \   \
	//   4   5   6
	const treeRoot = new Node(1);
	const b = new Node(2);
	const c = new Node(3);
	const d = new Node(4);
	const e = new Node(5);
	const f = new Node(6);
	treeRoot.left = b;
	treeRoot.right = c;
	b.left = d;
	b.right = e;
	c.right = f;

	return treeRoot;
}

function main() {
	console.log(createBT());
}

main();

File: DSA/100. Old Data/BinaryTree/BinaryTree.cpp
#include <iostream>
using namespace std;

class Node {
public:
  int value;
  Node *left;
  Node *right;
  Node(int value) {
    this->value = value;
    this->left = NULL;
    this->right = NULL;
  }
};

class BinaryTree {
public:
  Node *buildTree(Node *root) {
    cout << "Enter value : ";
    int value;
    cin >> value;
    root = new Node(value);

    if (value == -1) {
      return NULL;
    }

    cout << "For inserting in left : " << value << endl;
    root->left = buildTree(root->left);
    cout << "Enter data for inserting in right : " << value << endl;
    root->right = buildTree(root->right);

    return root;
  }
};

int main() {
  BinaryTree *bt = new BinaryTree();
  Node *root = bt->buildTree(NULL);

  // Tree Created
  //       1
  //      / \
  //     3   5
  //    / \   \
  //   7  15  17

  return 0;
}